### PR TITLE
Update the documentation for Decoded/Monad

### PR DESCRIPTION
**Pull request by gfontenot**

### CODE REVIEW

File: Argo/Types/Decoded/Monad.swift [hunk 1]
```diff
@@ -14,13 +16,15 @@ public func >>- <T, U>(x: Decoded<T>, @noescape f: T -> Decoded<U>) -> Decoded<U
...
 /**
-  flatMap a function over a `Decoded` value (left associative)
+  Conditionally map a function over a `Decoded` value, flattening the result.
 
-  - If the value is a failing case (`.TypeMismatch`, `.MissingKey`), the function will not be evaluated and this will return the value
-  - If the value is `.Success`, the function will be applied to the unwrapped value
+  - If the value is `.Failure`, the function will not be evaluated and this
+    will return `.Failure`.
+  - If the value is `.Success`, the function will be applied to the unwrapped
+    value.
```
Comment (gabebw) [inline Argo/Types/Decoded/Monad.swift]: ...and then it'll be re-wrapped, right?

Comment (gabebw) [inline Argo/Types/Decoded/Monad.swift]: I suppose the fact that it's `f: T -> Decoded<U>` makes that semi-clear, but I think it's nice to note.

Comment (gfontenot) [inline Argo/Types/Decoded/Monad.swift]: _we_ won't re-wrap it, the function will wrap it. I'm not sure how to communicate that.

Comment (gabebw) [inline Argo/Types/Decoded/Monad.swift]: Oh right, of course. I think this is fine then.

File: Argo/Types/Decoded/Monad.swift [hunk 2]
```diff
@@ -14,13 +16,15 @@ public func >>- <T, U>(x: Decoded<T>, @noescape f: T -> Decoded<U>) -> Decoded<U
...
 
-  - parameter x: A value of type `Decoded<T>`
   - parameter f: A transformation function from type `T` to type `Decoded<U>`
+  - parameter x: A value of type `Decoded<T>`
```
Comment (tonyd256) [inline Argo/Types/Decoded/Monad.swift]: the space below this is inconsistent with the other documentation PRs

File: Argo/Types/Decoded/Monad.swift [hunk 3]
```diff
@@ -29,6 +33,18 @@ public func -<< <T, U>(@noescape f: T -> Decoded<U>, x: Decoded<T>) -> Decoded<U
...
+  /**
+    Conditionally map a function over `self`, flattening the result.
+
+    - If `self` is `.Failure`, the function will not be evaluated and this will
+      return `.Failure`.
+    - If `self` is `.Success`, the function will be applied to the unwrapped
+      value.
+
+    - parameter f: A transformation function from type `T` to type `Decoded<U>`
+
```
Comment (tonyd256) [inline Argo/Types/Decoded/Monad.swift]: this extra line is inconsistent with the other doc PRs

**Pull request merged**